### PR TITLE
Disable User petition with setting

### LINKED ISSUE #227: Add a setting for an instance oriented toward organizations (no user publishing)
That could be the setting you would want for an instance reserved for an association / NGO / company / group where maybe you would not want individuals to create petitions in their own name but only in the name of the group.

This setting would change the behavior of the instance as follow:

* Users would not be able to create or publish a petition in their name
* They would only be able to do so on the dashboard of an organization they are a member of
* Already existing user-owned petitions would all be 'invisible' for visitors, as if they were unpublished.
* This setting could be set or unset at will with no irreversible effects.

This would possibly be used in addition with another setting that would restrict Organization creation to only admin (Django `super-user`) users. see #228

### PR DESCRIPTION
**Pull request by martinlehoux**
- Closes #227 
- Introduce new `DISABLE_USER_PETITION` setting
- Redirect to dashboard users trying to create a user petition with a warning
- Users still can use org petitions
- Also some linting on the base settings

**TODO :**
- [x] Document new setting

image

### CODE REVIEW

**Comment by martinlehoux** [comment]
there seem to be an unrelated error in the CI : `PermissionError: [Errno 13] Permission denied: '/var/lib/jenkins/workspace/Pytition_PR-237/pytition/mediaroot/julia/logo.png'`

**Comment by fallen** [comment]
About the CI issue,maybe try to rebase your branch on updated master and force push your branch?

**Comment by martinlehoux** [comment]
About the doc, I can add it to the config file with description and here https://pytition.readthedocs.io/en/latest/configuration.html ? Is that it ?

**Comment by fallen** [comment]
> About the doc, I can add it to the config file with description and here https://pytition.readthedocs.io/en/latest/configuration.html ? Is that it ?

Yes, in `settings.py` and in `doc/configuration.rst`

**Comment by fallen** [comment]
I don't understand why the CI is still red, but I guess that's not important, as long as when merged in master CI is green which I guess it will be!

**Comment by fallen** [comment]
I've just been thinking about something: we should also forbid users to transfer petitions to organizations while the `DISABLE_USER_PETITION` setting is set.
So that a previously created petition could not be transferred to circumvent the blocking.

That would be somewhere in there: https://github.com/pytition/Pytition/blob/master/pytition/petition/views.py#L1613

**Comment by martinlehoux** [comment]
Will work on that tonight

File: pytition/petition/views.py [hunk 1]
```diff
@@ -1605,6 +1605,9 @@ def transfer_petition(request, petition_id):
             except:
                 notFound = True
         if owner_type == "user":
+            if settings.DISABLE_USER_PETITION:
+                messages.error(request, _("Users are not allowed to be transfered a petition."))
```
Comment (fallen) [inline pytition/petition/views.py:1609]: I would say "Users are not allowed to transfer petitions to organizations on this instance"

Comment (martinlehoux) [reply pytition/petition/views.py:1609]:
```suggestion
                messages.error(request, _("Users are not allowed to transfer petitions to organizations on this instance."))
```

File: pytition/petition/tests/tests_PetitionTransfer.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+        response = self.client.post(url, data={"new_owner_type": "user", "new_owner_name": user.user.username}, follow=True)
+        self.assertEqual(response.status_code, 200)
+        user_petition = Petition.objects.get(id=user_petition.id)
+        self.assertEqual(user_petition.user, user)
+
+        with override_settings(DISABLE_USER_PETITION=True):
+            user_petition = Petition.objects.create(title="Petition 1", org=org)
+            response = self.client.post(url, data={"new_owner_type": "user", "new_owner_name": user.user.username}, follow=True)
+            self.assertContains(response, "Users are not allowed to be transfered a petition.")
```
Comment (martinlehoux) [inline pytition/petition/tests/tests_PetitionTransfer.py:96]:
```suggestion
            self.assertContains(response, "Users are not allowed to transfer petitions to organizations on this instance.")
```

**Pull request merged**